src/conways/conways_game.rs

```diff
@@ -4,13 +4,31 @@ use super::point::Point;
 
 pub struct ConwaysGame {
     alive_cells: HashSet<Point>,
+    height: u32,
+    width: u32,
 }
 
 impl ConwaysGame {
-    pub fn new(alive_cells: Vec<Point>) -> Self {
-        Self {
-            alive_cells: HashSet::from_iter(alive_cells),
+    pub fn new(alive_cells: Vec<Point>, height: u32, width: u32) -> Result<Self, String> {
+        for cell in &alive_cells {
+            if !Self::cell_is_in_range(cell, height, width) {
+                return Err(format!(
+                    "A cell was not in the specified range, game width: {}, game height: {}, cell x: {}, cell y: {}",
+                    width, height, cell.x_position, cell.y_position
+                ));
+            }
         }
+
+        Ok(Self {
+            alive_cells: HashSet::from_iter(alive_cells),
+            height,
+            width,
+        })
+    }
+
+    fn cell_is_in_range(cell: &Point, height: u32, width: u32) -> bool {
+        (0 <= cell.x_position && cell.x_position <= width as i32)
+            && (0 <= cell.y_position && cell.y_position <= height as i32)
```

Comment:
Both row == `HEIGHT` and column == `WIDTH` (for example point (50,50)) should not be valid, to fix this just change:
```suggestion
        (0 <= cell.x_position && cell.x_position < width as i32)
            && (0 <= cell.y_position && cell.y_position < height as i32)
```

Reply:
Ok, just adding your suggestion 🫡